app/helpers/session.rb

```diff
@@ -0,0 +1,46 @@
+def authorized?
+  unless User.find_by(id: session[:user_id])
+    p "User not found!"*100
+    return false
+  end
+  correct_password = User.find(session[:user_id]).password_hash
```

Comment:
You should not need to check the password here again. You should trust that if a user_id is in the session that it got there in a safe way.